### PR TITLE
improve execution step

### LINKED ISSUE #103: Make the `execute_workspace` step more reliable by having the prompt instruct it to navigate to the right path first

### PR DESCRIPTION
**Pull request by AntonOsika**
Fix #103

### CODE REVIEW

File: gpt_engineer/steps.py [hunk 1]
```diff
@@ -119,7 +122,9 @@ def execute_workspace(ai: AI, dbs: DBs):
         return messages
     print('Executing the code...')
     print()
-    subprocess.run(command, shell=True)
+
+    # Run the subprocess in dbs.workspace.path
+    subprocess.run('bash run.sh', shell=True, cwd=dbs.workspace.path)
```
Comment (patillacode) [inline gpt_engineer/steps.py:127]: Hey @AntonOsika 

I think I am missing something here, are we forcing the creation of a `run.sh` file somewhere?

Comment (jebarpg) [reply gpt_engineer/steps.py:127]: might need an environment flag to choose which shell to use for this. something like 

```python
import subprocess
import psutil
import os
#https://github.com/giampaolo/psutil
p = psutil.Process(os.getppid())
env = p.environ()
print(env)
if 'COMSPEC' in env:
    subprocess.run(f'cmd run.bat', shell=True, cwd=dbs.workspace.path)
if 'SHELL' in env:
    subprocess.run(f'{env['SHELL']} run.sh', shell=True, cwd=dbs.workspace.path) #{env['SHELL']}will return the shell like this '/bin/bash'

```

**Comment by patillacode** [comment]
With your permission @AntonOsika 
I am removing merged branches to avoid cluttering.

**Comment by AntonOsika** [comment]
Great. I'm used to that happening automatically 🙏 @patillacode

File: gpt_engineer/steps.py (hunk 1 above)
Comment (Rockin30) [reply gpt_engineer/steps.py:127]: Can I change this in my own build? Super unfamiliar with github and the like, but I'm trying to run gp-engineer through vscode terminal and conda env.

**Pull request merged**